lib/drush-node.js

```diff
@@ -104,5 +106,84 @@ Drush.exec = function (command, options) {
   return def.promise;
 };
 
+/**
+ * Spawn a drush command.
+ *
+ * @param {array} args
+ *   An array of arguments to append to the command, e.g. ['cc', 'drush'].
+ * @param {object} options
+ *   A hash of options to add to the command, can contain:
+ *     - alias: the drush alias, e.g. "@self" to execute the command with.
+ *     - simulate: boolean, simulates all relevant actions.
+ *     - uri: the URI of the drupal site to use.
+ *     - echo: text to echo to the drush command.
+ *     - cat: a file to cat to the drush command.
+ *
+ * @return Promise
+ */
+Drush.spawn = function (args, options) {
+  args = args || [];
+  options = options || {};
+  var def = new Deferred();
+  var pipeToDrush = function (command, pipeArgs) {
```

Comment:
If it can be, this should probably be defined outside the scope of this function. Since this is cli it doesn't really matter, but it's better practice and will help keep the code more readable.

It looks like the promise is the only thing that's shared between contexts so you could probably just have this method return the promise to be used by `Drush.spawn`.